libs/platform/ebpf_platform.h

```diff
@@ -124,6 +124,59 @@ extern "C"
     void
     ebpf_free(void* memory);
 
+    typedef enum _ebpf_page_protection
+    {
+        EBPF_PAGE_PROTECT_READ_ONLY,
+        EBPF_PAGE_PROTECT_READ_WRITE,
+        EBPF_PAGE_PROTECT_READ_EXECUTE,
+    } ebpf_page_protection_t;
+
+    typedef struct _ebpf_memory_descriptor ebpf_memory_descriptor_t;
+
+    /**
+     * @brief Allocate pages from physical memory and create a mapping into the
+     * system address space.
+     *
+     * @param length Size of memory to allocate (internally this gets rounded
```

Comment:
```suggestion
     * @param[in] length Size of memory to allocate (internally this gets rounded
```
Similarly please add [in,out] annotation throughout this file.